### PR TITLE
Prepare standalone package, step 2

### PR DESCRIPTION
**Pull request by nalimilan**
`mean` and `mean!` are now defined in Base.

Tests will fail until https://github.com/JuliaLang/julia/pull/46501 is merged.

### CODE REVIEW

File: src/Statistics.jl [hunk 1]
```diff
@@ -11,184 +11,11 @@ using LinearAlgebra, SparseArrays
 
 using Base: has_offset_axes, require_one_based_indexing
 
+using Base: mean!, mean
```
Comment (ararslan) [inline src/Statistics.jl:14]: Why is this necessary?

Comment (ararslan) [reply src/Statistics.jl:14]: We could do something like
```julia
if !isdefined(Base, :mean)
    export mean, mean!
end
```
which would make the intent a bit clearer, IMO, and could eventually be removed once the package requires a version of Julia that includes your `mean` PR.

Comment (nalimilan) [reply src/Statistics.jl:14]: Actually I assumed that we would merge this PR only after https://github.com/JuliaLang/julia/pull/46501 so that we don't need to support older versions. Anyway the package will only be usable on Julia 1.9 and above AFAICT.

Note that making the export conditional isn't enough to support older versions, we would also have to keep the method definitions.

Comment (nalimilan) [reply src/Statistics.jl]: Done.

**Comment by nalimilan** [comment]
Given that we have a chicken-and-egg problem to get CI to pass in Statistics.jl and in Julia, I've restored definitions for `mean` and `mean!`, making them conditional on `isdefined(Base, :mean)`. That way we can merge this here with CI passing, and then update https://github.com/JuliaLang/julia/pull/46501 to use the latest Statistics.jl and CI should pass there too. Then we can drop function definitions. OK?

**Comment by nalimilan** [comment]
I also dropped `mean` and `mean!` from the manual as that causes the Julia build to fail due to duplicate inclusion of docstrings.

**Comment by ararslan** [comment]
> `mean` and `mean!` are now defined in Base.

Big if true 😜

> I also dropped `mean` and `mean!` from the manual as that causes the Julia build to fail due to duplicate inclusion of docstrings.

Documenter complains about having the same docstring twice? I guess that makes sense since in most cases that's probably user error. Unfortunate that they need to be removed though because of that though. So in the period between the merging of this and merging the linked Julia PR, `mean`(`!`) won't appear in any manual builds?

File: src/Statistics.jl [hunk 2]
```diff
@@ -16,189 +16,189 @@ export cor, cov, std, stdm, var, varm, mean!, mean,
...
+
+    julia> mean([√1, √2, √3])
+    1.3820881233139908
+    ```
+    """
+    function mean(f, itr)
+        y = iterate(itr)
+        if y === nothing
+            return Base.mapreduce_empty_iter(f, +, itr,
+                                            Base.IteratorEltype(itr)) / 0
```
Comment (ararslan) [inline src/Statistics.jl:66]:
```suggestion
                                             Base.IteratorEltype(itr)) / 0
```

**Comment by nalimilan** [comment]
> Documenter complains about having the same docstring twice? I guess that makes sense since in most cases that's probably user error. Unfortunate that they need to be removed though because of that though. So in the period between the merging of this and merging the linked Julia PR, `mean`(`!`) won't appear in any manual builds?

No, because the master branch of Statistics will only be used after merging the Julia PR. Any changes made here will have zero effect on users as long as we don't bump the commit in Julia.

**Review by ararslan — APPROVED** [review]

**Comment by nalimilan** [comment]
Actually given that Statistics will remain an stdlib (at least for now), and simply become upgradable, `mean` isn't going to move to Base so this PR isn't needed anymore. https://github.com/JuliaStats/Statistics.jl/pull/148 reverts it.
